src/ip_notifier.cpp

```diff
@@ -130,7 +130,11 @@ struct ip_change_notifier_impl final : ip_change_notifier
 			[](SCNetworkReachabilityRef /*target*/, SCNetworkReachabilityFlags /*flags*/, void *info)
 			{
 				auto obj = static_cast<ip_change_notifier_impl*>(info);
-				obj->m_ios.post([obj]() { if (obj->m_cb) obj->m_cb(error_code()); });
+				if (!obj->m_cb) return;
```

Comment:
actually, looking at this I get a bit nervous about threads. the dispatch_queue you create, that's basically a thread pool, right? so, I suspect this callback will be called from some arbitrary thread. This is why you have to post it back onto the ``io_service`` message queue, to serialize.

The manipulation you do to ``obj`` below likely needs to happen *inside* the handler you post, right? Because this would likely cause a race condition (assuming my assumption about the threads is right)